src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1131,6 +1132,13 @@ static int client_send_discover(sd_dhcp_client *client) {
                         return r;
         }
 
+        if (!client->anonymize && client->rapid_commit) {
```

Comment:
Maybe, better to merge the previous condition for `client->anonymize`?

Reply:
Decided to conditionally set the client property in the setter based on anonymize. The anonymize value is known at creation time for the client, so this way the client->rapid_commit doesn't need to be checked w/ anonymize all the time. Would probably make sense for other boolean client properties to do the same, if there were any.